src/main/java/org/sagebionetworks/template/repo/RepositoryTemplateBuilderImpl.java

```
@@ -317,24 +319,40 @@ VelocityContext createSharedContext() {
 		String instance = config.getProperty(PROPERTY_KEY_INSTANCE);
 
 		// Describe the repository database.
-		results[0] = new DatabaseDescriptor().withResourceName(stack + instance + "RepositoryDB")
+		DatabaseDescriptor repoDbDescriptor = new DatabaseDescriptor().withResourceName(stack + instance + "RepositoryDB")
 				.withAllocatedStorage(config.getIntegerProperty(PROPERTY_KEY_REPO_RDS_ALLOCATED_STORAGE))
 				.withMaxAllocatedStorage(config.getIntegerProperty(PROPERTY_KEY_REPO_RDS_MAX_ALLOCATED_STORAGE))
 				.withInstanceIdentifier(stack + "-" + instance + "-db").withDbName(stack + instance)
 				.withInstanceClass(config.getProperty(PROPERTY_KEY_REPO_RDS_INSTANCE_CLASS))
 				.withDbStorageType(config.getProperty(PROPERTY_KEY_REPO_RDS_STORAGE_TYPE))
 				.withDbIops(config.getIntegerProperty(PROPERTY_KEY_REPO_RDS_IOPS))
 				.withMultiAZ(config.getBooleanProperty(PROPERTY_KEY_REPO_RDS_MULTI_AZ));
+		String repoSnapshotIdentifier = config.getProperty(PROPERTY_KEY_RDS_REPO_SNAPSHOT_IDENTIFIER);
+		if (! NOSNAPSHOT.equals(repoSnapshotIdentifier)) {
+			results[0] = repoDbDescriptor.withSnapshotIdentifier(repoSnapshotIdentifier);
+		} else {
+			results[0] = repoDbDescriptor;
+		}
 
+		String[] repoTableSnapshotIdentifiers = null;
+		if (repoSnapshotIdentifier != null) {
```

Maybe better to throw an exception instead of a "hidden" default